kai/incident_store_advanced.py

```diff
@@ -925,51 +972,143 @@ def accept_solution_from_id(
-        cmt_uri_origin,
-        cmt_uri_local,
-        "quarkus",
-        new_cmt_commit,
-        new_cmt_time,
-    )
+        # TODO:  This mapping data should be moved out of the code, consider moving to a config file
+        mapping = {
+            "eap-coolstore-monolith": "samples/sample_repos/eap-coolstore-monolith",
+            "ticket-monster": "samples/sample_repos/ticket-monster",
+            "kitchensink": "samples/sample_repos/kitchensink",
+            "helloworld-mdb": "samples/sample_repos/helloworld-mdb",
+            "bmt": "samples/sample_repos/bmt",
+            "cmt": "samples/sample_repos/cmt",
+            "ejb-remote": "samples/sample_repos/ejb-remote",
+            "ejb-security": "samples/sample_repos/ejb-security",
+            "tasks-qute": "samples/sample_repos/tasks-qute",
+            "greeter": "samples/sample_repos/greeter",
+        }
 
-    old_cmt_report = Report(
-        f"{BASE_PATH}/samples/analysis_reports/cmt/initial/output.yaml"
-    )
-    new_cmt_report = Report(
-        f"{BASE_PATH}/samples/analysis_reports/cmt/solved/output.yaml"
-    )
+        basedir = os.path.dirname(os.path.realpath(__file__))
+        parent_dir = os.path.dirname(basedir)
+        path = mapping.get(app_name, None)
+        return os.path.join(parent_dir, path)
 
-    psqlis.insert_and_update_from_report(old_cmt_application, old_cmt_report)
-    input("INSPECT!!")
-    psqlis.insert_and_update_from_report(new_cmt_application, new_cmt_report)
+    def get_app_variables(self, app_name: str):
+        # Path to the analysis_reports directory
+        analysis_reports_dir = "../samples/analysis_reports"
 
-    # for ruleset_name, ruleset_dict in report.get_report().items():
-    #   print(f"report_dict mapping: {type(ruleset_name)} -> {type(ruleset_dict)}")
+        # Check if the specified app folder exists
+        app_folder = os.path.join(analysis_reports_dir, app_name)
+        if not os.path.exists(app_folder):
+            print(
+                f"Error: {app_name} does not exist in the analysis_reports directory."
+            )
+            return None
+
+        # Path to the app.yaml file
+        app_yaml_path = os.path.join(app_folder, "app.yaml")
+
+        # Check if app.yaml exists for the specified app
+        if not os.path.exists(app_yaml_path):
+            print(f"Error: app.yaml does not exist for {app_name}.")
+            return None
 
-    #   for violation_name, violation_dict in ruleset_dict['violations'].items():
-    #     print(f"{violation_name=} {violation_dict.keys()}")
-    #   print()
+        # Load contents of app.yaml
+        with open(app_yaml_path, "r") as app_yaml_file:
+            app_data: dict = yaml.safe_load(app_yaml_file)
+
+        return app_data
+
+
+if __name__ == "__main__":
+
+    psqlis = PSQLIncidentStore(
+        config_filepath="database.ini",
+        config_section="postgresql",
+        emb_provider=EmbeddingNone(),
+        drop_tables=True,
```

Comment:
Probably don't want to drop the tables each time

Reply:
Makes sense. I added make target for loading data and we can change it via args